mapping/mapping.go

```diff
@@ -22,12 +22,21 @@ import (
 	"github.com/blevesearch/bleve/document"
 )
 
-// A Classifier is an interface describing any object
-// which knows how to identify its own type.
+// A Classifier is an interface describing any object which knows how to
+// identify its own type.  Alternatively, if a struct already has a Type
+// field or method in conflict, one can use BleveType instead.
 type Classifier interface {
 	Type() string
 }
 
+// A BleveClassifier is an interface describing any object which knows how
```

Comment:
I guess bleveClassifier interface is private, so lowercase.

Neat trick with the private interface, btw.